DeviceBridgeE2E/src/device-bridge.spec.ts

```diff
@@ -0,0 +1,205 @@
+    var invocationValueBody = JSON.parse(invocationValue.body);
+    t.is(invocationValueBody.status, "Disabled");
+    t.is(invocationValueBody.eventType, "ConnectionStatusChange");
+    t.is(invocationValueBody.deviceId, t.context.device.id);
+});
+
+test.serial('Test device desired property update callback', async t => {
+    // Create subscription to Azure Function
+    const callbackUrl = `${t.context.ctx.callbackUrl}&deviceId=${t.context.device.id}`;
+    await t.context.ctx.deviceBridgAPI.createDesiredPropertySubscription(t, t.context.device.id, callbackUrl)
+    await sleep(3000);
+
+    // Ensure get works
+    var response = await t.context.ctx.deviceBridgAPI.getDesiredPropertySubscription(t, t.context.device.id);
+    t.is(response.body.callbackUrl, callbackUrl)
+    t.is(response.body.status, "Running");
+    var propTestValue = "test";
+    await t.context.ctx.publicAPI.setProperties(t, t.context.device.id, {rwProp: propTestValue});
+    await sleep(3000);
+    var invocationValue = await t.context.ctx.deviceBridgAPI.getEcho(t, t.context.device.id);
+    var invocationValueBody = JSON.parse(invocationValue.body);
+    t.is(invocationValueBody.desiredProperties.rwProp, propTestValue);
+    t.is(invocationValueBody.eventType, "DesiredPropertyUpdate");
+    t.is(invocationValueBody.deviceId, t.context.device.id);
+
+    // Ensure connection deleted event when a sub deleted
+    await t.context.ctx.deviceBridgAPI.deleteDesiredPropertySubscription(t, t.context.device.id);
+    await sleep(2000);
+    var invocationValue = await t.context.ctx.deviceBridgAPI.getEcho(t, t.context.device.id);
+    var invocationValueBody = JSON.parse(invocationValue.body);
+    t.is(invocationValueBody.status, "Disabled");
+    t.is(invocationValueBody.eventType, "ConnectionStatusChange");
+    t.is(invocationValueBody.deviceId, t.context.device.id);
+});
+
+test.serial('Test device to cloud messaging', async t => {
+    const temperatureValue = 120;
+    await t.context.ctx.deviceBridgAPI.sendTelemetry(t, t.context.device.id, {
+        data: { temperature: temperatureValue },
+    });
+
+    var attempts = 5;
+    var telemetryResult = await t.context.ctx.publicAPI.getLatestDeviceTelemetry(
+        t,
+        t.context.device.id,
+        'temperature'
+    );
+    // Multiple attempts as it may take time for telemetry to come through
+    for (var i = 0; i < attempts; i++) {
+        if (telemetryResult != {} && telemetryResult.value != undefined) {
+            break;
+        }
+        await new Promise(r => setTimeout(r, 2000));
+        telemetryResult = await t.context.ctx.publicAPI.getLatestDeviceTelemetry(
+            t,
+            t.context.device.id,
+            'temperature'
+        );
+    }
+    t.is(String(telemetryResult.value), String(temperatureValue));
+});
+
+test.serial('Test reported properties and twin', async t => {
+    var reportedPropertiesBody = {
+        "patch": {
+            "rwProp": "testValue",
```

Comment:
Also use a random value here, tom make sure we're not reading the result of a previous execution.